core/graph/src/main/java/uk/gov/gchq/gaffer/graph/hook/NamedViewResolver.java

```diff
@@ -0,0 +1,90 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.gaffer.graph.hook;
+
+import uk.gov.gchq.gaffer.data.elementdefinition.view.NamedView;
+import uk.gov.gchq.gaffer.named.operation.cache.exception.CacheOperationFailedException;
+import uk.gov.gchq.gaffer.operation.Operation;
+import uk.gov.gchq.gaffer.operation.OperationChain;
+import uk.gov.gchq.gaffer.operation.Operations;
+import uk.gov.gchq.gaffer.operation.graph.OperationView;
+import uk.gov.gchq.gaffer.store.Context;
+import uk.gov.gchq.gaffer.store.operation.handler.named.cache.NamedViewCache;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A {@link GraphHook} to resolve named operations.
+ */
+public class NamedViewResolver implements GraphHook {
+    private final NamedViewCache cache;
+
+    public NamedViewResolver() {
+        cache = new NamedViewCache();
+    }
+
+    public NamedViewResolver(final NamedViewCache cache) {
+        this.cache = cache;
+    }
+
+    @Override
+    public void preExecute(final OperationChain<?> opChain, final Context context) {
+        resolveViewsInOperations(opChain);
+    }
+
+    @Override
+    public <T> T postExecute(final T result, final OperationChain<?> opChain, final Context context) {
+        return null;
+    }
+
+    @Override
+    public <T> T onFailure(final T result, final OperationChain<?> opChain, final Context context, final Exception e) {
+        return null;
+    }
+
+    private void resolveViewsInOperations(final Operations<?> operations) {
+        final List<Operation> updatedOperations = new ArrayList<>(operations.getOperations().size());
+        for (final Operation operation : operations.getOperations()) {
+            if (operation instanceof OperationView) {
+                if (((OperationView) operation).getView() instanceof NamedView) {
+                    final NamedView resolvedNamedView = resolveNamedViewInOperation((NamedView) ((OperationView) operation).getView());
+
+                    final NamedView mergedNamedView = new NamedView.Builder().name(resolvedNamedView.getName()).merge(resolvedNamedView).merge(((OperationView) operation).getView()).build();
+                    ((OperationView) operation).setView(mergedNamedView);
+                }
+            }
+            updatedOperations.add(operation);
+        }
+        operations.getOperations().clear();
+        operations.getOperations().addAll((List) updatedOperations);
+    }
+
+    private NamedView resolveNamedViewInOperation(final NamedView opView) {
+        NamedView.Builder newNamedView;
+        try {
+            newNamedView = new NamedView.Builder().name(opView.getName()).merge(cache.getNamedView(opView.getName()));
```

Comment:
I don't think you need to create a new NamedView object here. Can you create a View object instead?

Reply:
I need to keep the name, so when the merge is run it will add the name to the mergedNamedViewNames, otherwise it will still work but the user won't be able to keep track of what the current NamedView consists of.